overlord/snapstate/backend/copydata_test.go

```diff
@@ -64,7 +64,20 @@ version: 2.0
 )
 
 func (s *copydataSuite) TestCopyData(c *C) {
-	homedir := filepath.Join(s.tempdir, "home", "user1", "snap")
+	for _, t := range []struct {
+		snapDir string
+		opts    *dirs.SnapDirOptions
+	}{
+		{snapDir: dirs.UserHomeSnapDir, opts: nil},
+		{snapDir: dirs.UserHomeSnapDir, opts: &dirs.SnapDirOptions{}},
+		{snapDir: dirs.HiddenSnapDataHomeDir, opts: &dirs.SnapDirOptions{HiddenSnapDataDir: true}}} {
+
+		s.testCopyData(c, t.snapDir, t.opts)
```

Comment:
I suppose we do have this snapstate test, but it would be nice to see a higher level test which checks that we get the option from the experimental feature setting and based on that copy the right files